mgmtsystem/scvmm.py

```diff
@@ -103,26 +103,82 @@ def wait_vm_stopped(self, vm_name, num_sec=300):
+        """.format(vm_name=vm_name, vm_new_name=vm_new_name)
+        self.logger.info(" Renaming SCVMM VM `{}` to  `{}`"
+            .format(vm_name, vm_new_name))
+        self.run_script(script)
+
+    def delete_vm(self, vm_name, host):
+        if not self.is_vm_stopped(vm_name) and self.vm_status(vm_name) not in self.STATES_FAILED:
+            self.stop_vm(vm_name)
+            self.wait_vm_stopped(vm_name)
+        script = """
+        $VM = Get-SCVirtualMachine -Name \"{vm_name}\" -VMMServer $scvmm_server
+        Remove-SCVirtualMachine -VM $VM
+        """.format(vm_name=vm_name, host=host)
+        self.logger.info(" Deleting SCVMM VM `{}`"
+            .format(vm_name))
+        self.run_script(script)
+
+    def delete_template(self, template):
+        script = """
+        $Template = Get-SCVMTemplate -Name \"{}\" -VMMServer $scvmm_server
+        Remove-SCVMTemplate -VMTemplate $Template -Force
+        """.format(template)
+        self.logger.info(" Removing SCVMM VM `{}`"
+            .format(template))
+        self.run_script(script)
 
     def restart_vm(self, vm_name):
         self._do_vm(vm_name, "Reset")
 
     def list_vm(self, **kwargs):
         data = self.run_script(
-            "Get-SCVirtualMachine -All -VMMServer $scvmm_server | "
-            "where { $_.MarkedAsTemplate -eq $FALSE } | convertto-xml -as String")
+            "Get-SCVirtualMachine -All -VMMServer $scvmm_server |"
+            "Select name | ConvertTo-Xml -as String")
         return etree.parse(StringIO(data)).getroot().xpath("./Object/Property[@Name='Name']/text()")
 
+    def all_vms(self, **kwargs):
+        vm_list = []
+        data = self.run_script("""
+            $outputCollection = @()
+            $VMs = Get-SCVirtualMachine -All -VMMServer $scvmm_server |
+            Select VMId, Name, StatusString
+            $NetAdapter = Get-SCVirtualNetworkAdapter -VMMServer $scvmm_server -All |
+            Select ID, Name, IPv4Addresses
+            #Associate objects
+            $VMs | ForEach-Object{
+                $vm_object = $_
+                $ip_object = $NetAdapter | Where-Object {$_.Name -eq $vm_object.Name}
+
+                #Make a combined object
+                $outObj = "" | Select VMId, Name, StatusString, IPv4
+                $outObj.VMId = $vm_object.VMId
+                $outObj.Name = $vm_object.Name
+                $outObj.StatusString = $vm_object.StatusString
+                $outObj.IPv4 = if($ip_object.IPv4Addresses){$ip_object.IPv4Addresses} else {"None"}
+                #Add the object to the collection
+
+                $outputCollection += $outObj
+            }
+            $outputCollection | ft -HideTableHeaders
+            """)
+        lines = iter(data.splitlines())
+        for vm in lines:
+            fields = vm.split()
+            vm_data = fields[0], fields[1], fields[2], fields[3]
+            vm_list.append(vm_data)
```

Comment:
2 more things need to be done here:

At the top:
```python
from .base import VMInfo   # I think, or maybe mgmtsystem.base, it depends
```

And here:
```python
vm_data = fields[0], fields[1], fields[2], None if fields[3] == 'None' else fields[3]
vm_list.append(VMInfo(*vm_data))
```

I wrap the result in a namedtuple for easy access and also the None cannot be a string but a real None. And it definitely should not return a string, just the `vm_list`